optuna/visualization/_timeline.py

```diff
@@ -79,11 +79,62 @@ def objective(trial):
     return _get_timeline_plot(info)
 
 
+def _get_max_datetime_complete(study: Study) -> datetime.datetime:
+    max_run_duration = _get_max_run_duration(study)
+    if _is_running_trials_in_study(study, max_run_duration):
+        return datetime.datetime.now()
+
+    return max(
+        [t.datetime_complete for t in study.trials if t.datetime_complete is not None],
+        default=datetime.datetime.now(),
+    )
+
+
+def _get_max_run_duration(study: Study) -> datetime.timedelta | None:
+    max_run_duration = None
+    for t in study.trials:
+        if t.datetime_complete is None or t.datetime_start is None:
+            continue
+        time_delta = t.datetime_complete - t.datetime_start
+        if max_run_duration is not None:
+            max_run_duration = max(max_run_duration, time_delta)
+        else:
+            max_run_duration = time_delta
+
+    return max_run_duration
+
+
+def _is_running_trials_in_study(study: Study, max_run_duration: datetime.timedelta | None) -> bool:
+    if max_run_duration is None and [TrialState.RUNNING == t.state for t in study.trials]:
```

Comment:
```suggestion
    if max_run_duration is None and study.get_trials(states=(TrialState.RUNNING,), deepcopy=False):
```

Reply:
This can be expressed more concisely.
```python
if max_run_duration is None:
    return len(study.get_trials(states=(TrialState.RUNNING,), deepcopy=False)) > 0
```